examples/chip-tool/commands/common/Commands.h

```diff
@@ -23,6 +23,15 @@
 
 #include <controller/CHIPDeviceController_deprecated.h>
 
+class ChipToolPersistentStorageDelegate : public chip::Controller::PersistentStorageDelegate
+{
+    void SetDelegate(chip::Controller::PersistentStorageResultDelegate * delegate) override {}
+    void GetKeyValue(const char * key) override {}
+    CHIP_ERROR GetKeyValue(const char * key, char * value, uint16_t & size) override { return CHIP_NO_ERROR; }
```

Comment:
should it be 'NOT FOUND' or something like that? 

Also nitpicking now that I see the overrides:

- setters should really return an error code and so should deleters (what if write to flash fails)
- a void GetKeyValue(const char*key) is confusing to me. What does it do?

... maybe we can address those as we build up real storage. @rgoliver

Reply:
Looked up the docs. `GetKeyValue` should be `SendKeyValueToDelegate`.

I think we should decide on either delegate or sync calls and use them as such.